blog-post/blog-post.md

```diff
@@ -0,0 +1,522 @@
+end that will:
+
+- introduce powerful new features (generics, traits, algebraic datatypes, pattern matching, comptime)
+- provide a strong foundation for compiler correctness as we continue to extend the language in the future
+- empower library authors and support a community-driven process for language evolution
+- expand the capabilities of verification and analysis tooling
+
+In this post we will present our current progress and give you some idea of our vision for the
+future of the language. It is important to understand that the version presented here is not final,
+changes can and will be made before release. We are presenting our current progress now in order to
+receive feedback and to allow for course correction if necessary.
+
+As a summary of the current state of things: We have a working prototype for the new language
+version. All examples in the post typecheck and produce executable code. The syntax is far
+from final, and extensive changes should be expected before release. Much of the core type theory is
+stable, but we want to add at least compile time evaluation and modules.
+
+## A Note on Syntax
+
+Most of the work to date has been focused on the design and implementation of the type system, and
+associated code generation pipeline down to Yul. In order to avoid getting bogged down in
+bikeshedding around syntax, and with the desire to validate our core ideas with a working
+implementation as soon as possible, we moved ahead with a provisional syntax. You can expect
+extensive syntax changes before release. Our current intention is to match the syntax of Classic
+Solidity as much as possible. For any new syntax, you should expect that the final version will feel
+closer to more familiar languages like TypeScript or Rust than it does right now.
+
+We made the decision to share before we have a finalized syntax because we are both excited to
+share our progress, and wish to gather feedback sooner rather than later.
+
+## New Language Features
+
+Core Solidity takes ideas from pure functional programming languages (e.g. Haskell, Lean), as well
+as modern systems languages (e.g. Rust, Zig). The core new features that we plan to add are:
+
+- Generics / parametric polymorphism
+- Traits / typeclasses
+- Algebraic datatypes (a.k.a sum / product types) and pattern matching
+- Higher order functions
+- Type inference
+- Compile time evaluation
+
+We think that taken together these core primitives will enable developers to produce stronger
+abstractions, write more modular and reusable code, leverage the type system to enforce core
+safety properties.
+
+We will continue to support the kind of low level access to the EVM that is often required by
+production implementations: assembly will remain a core primitive, and we will support calling
+functions from the high level language in assembly directly. Users will be able to disable the built
```

Comment:
Calling functions from high-level language in assembly is not something you can currently do in Classic Solidity, so IMO it should be presented as a new feature. Right now it sounds as if it was already in and you just planned to keep it.